app/javascript/packs/document-capture.jsx

```diff
@@ -21,20 +22,22 @@ const device = {
     /ip(hone|ad|od)|android/i.test(window.navigator.userAgent),
 };
 
-const appRoot = document.getElementById('document-capture-form');
-const isLivenessEnabled = appRoot.hasAttribute('data-liveness');
-render(
-  <AcuantProvider
-    credentials={getMetaContent('acuant-sdk-initialization-creds')}
-    endpoint={getMetaContent('acuant-sdk-initialization-endpoint')}
-  >
-    <I18nContext.Provider value={i18n.strings}>
-      <AssetContext.Provider value={assets}>
-        <DeviceContext.Provider value={device}>
-          <DocumentCapture isLivenessEnabled={isLivenessEnabled} />
-        </DeviceContext.Provider>
-      </AssetContext.Provider>
-    </I18nContext.Provider>
-  </AcuantProvider>,
-  appRoot,
-);
+loadPolyfills(['fetch']).then(() => {
```

Comment:
It occurs to me that an alternative might be to load the polyfills at the time and place they're needed.

Example:

```js
async function upload(payload) {
  await loadPolyfills(['fetch']);
  window.fetch( /* ... */ );
```

A few things I observe from this, if it's a better alternative:

- It probably makes sense for `loadPolyfills` to be changed to singular `loadPolyfill`, since the more common case would be need for a single feature. If multiple are needed, it's still easy enough to use `Promise.all` from the consuming end.
- It could become repetitive if needed multiple times, which also leads to another point: It could be good to centralize this somehow, like an internal `fetch` wrapper that manages the polyfill.
   - Question remains where it might be best for this wrapper to exist, if internal to `document-capture`, or pulled into the new `@18f/identity-polyfill`, or something entirely different (`@18f/identity-fetch` even, though at that point, we may be getting a little carried away 😅 ).

Reply:
Preloading makes sense to me, typically older browsers and older machines will not have the things we're poly, right? So then on-demand loading of new code in the middle of the flow would add a little pause and make things feel even slower?

By preloading I think we can minimize that

Reply:
> Preloading makes sense to me, typically older browsers and older machines will not have the things we're poly, right? So then on-demand loading of new code in the middle of the flow would add a little pause and make things feel even slower?
> 
> By preloading I think we can minimize that

Ah, you're totally right, and I completely overlooked that aspect. There might be some benefit to the approach I mentioned previously if there a likelihood that the polyfill might not be needed at all. In this flow, since we're pretty confident the user will eventually submit the form, it's probably better to do this up-front as you mention. 

I guess either way, the "cost" (load) will need to be paid, whether that's up front or at the time of upload. Doing it up-front has some downsides as well, since there's already going to be other things which need to be loaded at the same time. But I think it's pretty expected at the initial load, and not so much later after page load.

The Acuant behavior is a good example of a middle-ground option, where the load starts immediately, but the application is still allowed to render. It's more complicated, since you need to handle cases where the script may or may not be ready by the time it's needed (#4063). It's a bit more justified for Acuant, since it affects all users, and is a much larger script to be loaded. For something like polyfills, I think it's generally acceptable that legacy browsers may endure a less-than-perfectly-optimized implementation.

Regarding the note "some benefit to the approach [...] if there a likelihood that the polyfill might not be needed at all", `loadPolyfills` should be able to support this as well. If we ever decide to take advantage of it, it'd probably make sense to enhance it further to avoid multiple loads of the same polyfill (internal [`once`-like behavior](https://lodash.com/docs/4.17.15#once)).